### ISSUE TITLE
RanibowKit Button component not updating the balance

### ISSUE THREAD

**Issue by technophile-04** [scaffold-eth/se-2#30]
## Description 
![Untitled design](https://user-images.githubusercontent.com/80153681/196988856-1513814e-5b6b-43d6-bc88-7b0f67ac359b.gif)

### Current Behaviour :
RainbowKit Button doesn't seem to update/re-fetch the connected address `Balance` after some interval. 
As in the above video when you grab funds from faucet(aka `0th` account of hardhat), the native scaffold-eth `Balance` component seems to show the updated balance of the `0th` account of hardhat, but the connected **address Balance** is not re-fetched/updated by **RainbowKit component**. 

### Expected Behaviour :
Similar to the native scaffold-eth `Balance` component, the RainbowKit component should also re-fetch the balance of the connected address after a small interval at least when you  are connected to the local hardhat chain

**Comment by carletex** [comment]
@Naim-Bijapure made a fix attempt here: https://github.com/scaffold-eth/se-2/pull/32

Let's find out what is the best way to go about it.

As Shiv said:
> Similar to the native scaffold-eth-1 Balance component, the RainbowKit component should also re-fetch the balance of the connected address after a small interval at least when you are connected to the local hardhat chain

As we do in SE1:
- Local chain: fetch balance onBlock
- External chain: fetch balance every X seconds

We might also want to have  "refetch" function that you can call if you don't to wait the X secs.

**Comment by carletex** [comment]
It seems that it's nos supported by default on Rainbow kit. See https://github.com/rainbow-me/rainbowkit/discussions/626

I tried a similar workaround as @Naim-Bijapure suggested:

```js
  const [connectKey, setConnectKey] = useState(0);

  useEffect(() => {
    const intervalId = setInterval(() => {
      setConnectKey(Math.floor(Math.random() * 1000));
    }, 2000);

    return () => clearInterval(intervalId);
  }, []);

  return (
      <ConnectButton
        // .....
        key={connectKey}
      />
  );
```

This works, but it creates a re-rendering blip that we don't want (only the second one is capture by the video):

https://user-images.githubusercontent.com/2486142/199526616-8f5b7e66-dcdc-4637-9dc0-d3ccea115046.mp4

I'm wondering if the solution is to implement a [Custom Connect Button](https://www.rainbowkit.com/docs/custom-connect-button)

**Comment by carletex** [comment]
Fixed on #43
